docker_test.go

```diff
@@ -83,3 +83,23 @@ func TestPortMappingLabels(t *testing.T) {
 	assert.NotNil(t, pm.Labels)
 	assert.Equal(t, 0, len(*pm.Labels))
 }
+
+func TestExternalVolumes(t *testing.T) {
+	container := NewDockerApplication().Container
+
+	container.Volume("", "cp", "RW")
+	ev := (*container.Volumes)[0].ExternalVolume("myVolume", "dvdi")
+
+	ev.AddOption("prop", "pval")
+	ev.AddOption("dvdi", "rexray")
+
+	assert.Equal(t, (*container.Volumes)[0].ContainerPath, "cp")
+	assert.Equal(t, (*container.Volumes)[0].External.Name, "myVolume")
+	assert.Equal(t, (*container.Volumes)[0].External.Provider, "dvdi")
+	assert.Equal(t, 1, len(*container.Volumes))
```

Comment:
This belongs to a test dedicated to validate that `Volume()` works correctly. Unfortunately, we didn't implement such a test when we added `Volume()`. :-(

I'll leave it up to you to either create `TestVolume` and move the length check there, or just drop it in your test.